### PR TITLE
workflows: add scorecard and codeql

### PR DESCRIPTION
**Pull request by tkatila**
Tweak token permissions and add scorecard badge.

### CODE REVIEW

**Comment by tkatila** [comment]
Hold this. I'll rewrite the structure so that it is more inline with the overall file structure.

File: .github/workflows/lib-scorecard.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  analysis:
+    name: Analysis
+    runs-on: ubuntu-latest
```
Comment (mythi) [inline .github/workflows/lib-scorecard.yaml:12]: we have followed a locked version:
```suggestion
    runs-on: ubuntu-22.04
```

(applies to other `ubuntu-latest` too)

**Review by mythi — COMMENTED** [review]
was there a badge for dependabot too?

**Comment by tkatila** [comment]
> was there a badge for dependabot too?

I couldn't find instructions for it. And it seems it might be broken at the moment: https://github.com/dependabot/dependabot-core/issues/1912

**Review by mythi — APPROVED** [review]

**Pull request merged**